### PR TITLE
Define common arcadia interface to align UX and API development

### PR DESCRIPTION
**Pull request by t-rufang**

### CODE REVIEW

File: Utils/hdinsight-node-common/src/com/microsoft/azure/projectarcadia/common/ArcadiaSparkComputeManager.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    private static class LazyHolder {
+        static final ArcadiaSparkComputeManager INSTANCE = new ArcadiaSparkComputeManager();
+    }
+
+    public static ArcadiaSparkComputeManager getInstance() {
+        return LazyHolder.INSTANCE;
+    }
+
+    @Override
+    public ImmutableSortedSet<? extends IClusterDetail> getClusters() {
```
Comment (lcadzy) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/projectarcadia/common/ArcadiaSparkComputeManager.java]: >getClusters [](start = 56, length = 11)

fetchClusters

File: Utils/hdinsight-node-common/src/com/microsoft/azure/projectarcadia/common/ArcadiaSparkComputeManager.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        return ImmutableSortedSet.of();
+    }
+
+    @Override
+    public ClusterContainer refresh() {
+        return null;
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/projectarcadia/common/ArcadiaSparkComputeManager.java]: NotNull, just `return this;`

File: Utils/hdinsight-node-common/src/com/microsoft/azure/projectarcadia/common/ArcadiaWorkSpace.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ */
+
+package com.microsoft.azure.projectarcadia.common;
+
+import com.google.common.collect.ImmutableSortedSet;
+import com.microsoft.azure.hdinsight.sdk.cluster.ClusterContainer;
+import com.microsoft.azure.hdinsight.sdk.cluster.IClusterDetail;
+import rx.Observable;
+
+public class ArcadiaWorkSpace implements ClusterContainer {
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/projectarcadia/common/ArcadiaWorkSpace.java]: Duplicate with `ArcadiaSparkComputeManager`?

File: Utils/hdinsight-node-common/src/com/microsoft/azure/projectarcadia/common/ArcadiaSparkCompute.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    @Override
+    public String getTitle() {
+        return null;
+    }
+
+    @Override
+    public String getConnectionUrl() {
+        return null;
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/projectarcadia/common/ArcadiaSparkCompute.java]: Notnull here

**Pull request merged**